pkg/mock/userdata/userdata.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package userdata
+
+import (
+	"log"
+	"net/http"
+	"reflect"
+
+	cfg "github.com/aws/amazon-ec2-metadata-mock/pkg/config"
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/mock/imdsv2"
+	"github.com/aws/amazon-ec2-metadata-mock/pkg/server"
+)
+
+var (
+	supportedPaths = make(map[string]interface{})
+	response       interface{}
+	// ServicePath defines the userdata service path
+	ServicePath = "/latest/user-data"
+)
+
+// Handler processes http requests
+func Handler(res http.ResponseWriter, req *http.Request) {
+	log.Println("Received request to mock userdata:", req.URL.Path)
+
+	if val, ok := supportedPaths[req.URL.Path]; ok {
+
+		response = val
+	} else {
+		response = "Something went wrong with: " + req.URL.Path
+	}
+	server.FormatAndReturnOctetResponse(res, response.(string))
+
+}
+
+// RegisterHandlers registers handlers for userdata paths
+func RegisterHandlers(config cfg.Config) {
+	server.HandleFunc("/latest/api/token", imdsv2.GenerateToken)
```

Comment:
not needed here; please remove

Reply:
Can you clarify the ask here.  I'm pretty sure RegisterHandlers is needed here.

Reply:
Correct, RegisterHandlers is needed. please remove: `server.HandleFunc("/latest/api/token", imdsv2.GenerateToken)` because `static.go` already registers this path for all mocks

Reply:
https://github.com/aws/amazon-ec2-metadata-mock/pull/171/commits/bcb7fc790d45b5a31dbd6ccfe9b8345b2828c76a